python/utilities/submit_chain/submit_chain.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+"""Submits a chain to a list of logs."""
+
+import base64
+import hashlib
+import sys
+
+import json
+import logging
+import gflags
+
+from ct.client import log_client
+from ct.crypto import cert
+from ct.crypto import error
+from ct.crypto import pem
+from ct.crypto import verify
+from ct.proto import client_pb2
+from ct.serialization import tls_message
+
+FLAGS = gflags.FLAGS
+
+gflags.DEFINE_string("log_list", None, "List of logs to submit to.")
+gflags.DEFINE_string("chain", None, "Certificate chain to submit (PEM).")
+gflags.DEFINE_string("log_schema", "http", "Log schema (http/https)")
+gflags.DEFINE_string("output", None, "output file for sct_list")
+gflags.MarkFlagAsRequired("log_list")
+gflags.MarkFlagAsRequired("chain")
+gflags.MarkFlagAsRequired("output")
+
+def _read_ct_log_list(log_list_file):
+    """Parses the log list JSON, returns a log url to key map."""
+    try:
+        log_list_json = json.loads(log_list_file)
+        log_url_to_key = {}
+        for log_info in log_list_json['logs']:
+            log_url_to_key[FLAGS.log_schema + '://' + log_info['url']] = (
```

Comment:
You might just require that the file already contains full URLs including scheme and avoid doing this.
Then you avoid the possibility of ending up with http://http://my.log.com when the user didn't realise their file was not supposed to include the scheme...